### PR TITLE
Add a FractionalOffset alignment parameter to BackgroundImage

### PR DESCRIPTION
**Pull request by tonygentilcore**
/cc @abarth

### CODE REVIEW

File: packages/flutter/lib/src/painting/box_painter.dart [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import 'dart:math' as math;
 import 'dart:ui' as ui;
 
+import 'package:flutter/rendering.dart' show FractionalOffset;
```
Comment (abarth) [inline packages/flutter/lib/src/painting/box_painter.dart]: This import is a layering inversion.  Instead, we should just move the definition of FractionalOffset into the painting layer.

**Comment by abarth** [comment]
Looks good except for the layering inversion.

**Comment by tonygentilcore** [comment]
Thanks, ptal

**Comment by abarth** [comment]
LGTM

**Pull request merged**